### PR TITLE
Add vocabsearch to "All Vocabs" screen

### LINKED ISSUE #108: Search Entries in the Packages
As a user of vocascan-fronend, I wish to have the possibility to search the entries of the packages I have entered in my account. This would ease the continuous enhancement of own vocabularies, for example to see if the entry is already entered, or to follow a previous example.

The search should be possible on the page (or level) of packages, and inside by at the separated packages. This give the user to chance to filter the search, either in all the packages, or in specific on

The result should be clickable, i.e. show the editing page, as it is now in browsing the entries of a package.

### PR DESCRIPTION
**Pull request by Tracer1337**
| Status  | Type  |
| :---: | :---: |
| :white_check_mark: Ready | Feature |

## Description

Added vocab search to "All Vocabs" screen.

Bug on server: Only matching translations are returned which results in incomplete vocabulary cards (look for "Peanuts, Treenuts" in the GIF)

## Screenshots / GIFs (if appropriate):

Animation

## Resolves

fixes #108

### CODE REVIEW

**Comment by luwol03** [comment]
Thank you for your pull request.
  Please remove your translations. Change only the `en/default.json` if necessary. 
  If this pull request is merged into the [`experimental`](https://github.com/vocascan/vocascan-frontend/tree/experimental) branch, it can be translated via [localazy](https://localazy.com/p/vocascan).
  
  Please remove following files from pull request changes.
  ```
  - src/i18n/locales/de/default.json
  ```
  (posted on behalf of @VocascanBot )

**Comment by Tracer1337** [comment]
> Thank you for your pull request. Please remove your translations. Change only the `en/default.json` if necessary. If this pull request is merged into the [`experimental`](https://github.com/vocascan/vocascan-frontend/tree/experimental) branch, it can be translated via [localazy](https://localazy.com/p/vocascan).
> 
> Please remove following files from pull request changes.
> 
> ```
> - src/i18n/locales/de/default.json
> ```
> 
> (posted on behalf of @VocascanBot )

Its removed

File: src/screens/Library/AllVocabs/AllVocabs.jsx [hunk 1]
```diff
@@ -27,13 +28,14 @@ const AllVocabs = () => {
...
   const [data, setData] = useState([]);
+  const [search, setSearch] = useState("");
   const [currentVocab, setCurrentVocab] = useState(null);
   const [showVocabModal, setShowVocabModal] = useState(false);
   const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] =
     useState(false);
 
   const fetchVocabs = useCallback(() => {
-    getGroupVocabulary(groupId).then((response) => {
+    getGroupVocabulary(groupId, search).then((response) => {
```
Comment (luwol03) [inline src/screens/Library/AllVocabs/AllVocabs.jsx:38]: can you please use the `useDebounce` hook to not over query the server

Comment (Tracer1337) [reply src/screens/Library/AllVocabs/AllVocabs.jsx]: Done

**Review by luwol03 — APPROVED** [review]
LGTM @noctera can we merge this?

**Comment by noctera** [comment]
Yes we can merge it

**Pull request merged**